File: 프로그래머스/lv1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    idx = 0
    while idx<len(section):
        cur = section[idx]
        s = idx
        for i in range(s,len(section)):
            n = section[i]
            if n >= cur+m: break
            idx = i +1
        answer +=1
    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int idx = 0;
        while(idx < section.length) {
            int cur = section[idx];
            int s = idx;
            for (int i = s; i<section.length; i++){
                n = section[i];
                if(n >= cur+m) break;
                idx = i+1;
            }
            answer++;
        }
        return answer;
    }
}